clang/lib/Basic/Targets/SPIR.h

```diff
@@ -70,11 +60,9 @@ class LLVM_LIBRARY_VISIBILITY SPIRTargetInfo : public TargetInfo {
     TLSSupported = false;
     VLASupported = false;
     LongWidth = LongAlign = 64;
-    if (Triple.getEnvironment() == llvm::Triple::SYCLDevice) {
-      AddrSpaceMap = &SYCLAddrSpaceMap;
-    } else {
-      AddrSpaceMap = &SPIRAddrSpaceMap;
-    }
+    AddrSpaceMap = (Triple.getEnvironment() == llvm::Triple::SYCLDevice)
```

Comment:
Not sure how to interpret this...
Do you want to me revert it? Add to the PR description? No actions implied?

Reply:
Yeah, sorry for not being clear.  I guess the answer is: It depends on where this lands.  I'd rather you do a separate commit for this formatting change here, but if you're submitting to llvm community, I'd rather the ternary version just be part of the patch.

Reply:
This patch is to SYCL project.
First, I'd like to commit it here and see if it breaks anything. So far internal testing do not exposed any issues.
In addition to that I'd like to get high-level feedback on the approach before moving to the LLVM project.

WRT using different address space map for SYCL. I don't know if we break anything by mapping "default" address space to "4", which SPIR-V converter interprets as generic. It might that other programming models targeting SPIR rely on existing mapping.
In theory OpenCL is not supposed to use "default" address space, so it should be okay to use SYCLAddrSpaceMap in all cases.
As it was a separate patch, I suggest doing this experiment separately. Does it sounds okay to you?

Tagging other SYCL implementers: @Naghasan, @keryell, @illuhad,

Reply:
Having lang address space in target information seems strange. Ideally the IR should be emitted as language agnostic as possible. What are you trying to achieve with this separate address space map?

Reply:
> Having lang address space in target information seems strange. Ideally the IR should be emitted as language agnostic as possible. What are you trying to achieve with this separate address space map?

I'm trying to map "pointer w/o address space qualifiers" to "generic" address space. Currently it's mapped to "private", which doesn't seems right. As mentioned in the previous comment if all other languages do not rely on that behavior, I can remove "language dependent" customization.

Reply:
Well OpenCL 2.0 does exactly the same. All pointer are in generic address space. Why don't you just add generic on AST while parsing the types?

Reply:
@erichkeane answered this in this comment: https://github.com/intel/llvm/pull/1039/files#r370236039.